### ISSUE TITLE
Test for TAILs on sources with empty frontiers

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#3468]
See #3465—just tracking getting a test for this while it's still fresh in everyone's minds.

@JLDLaughlin happy to help try to induce this situation. If it's just a matter of getting a source with an empty upper frontier, I think possibly trying to tail a materialized constant view (`CREATE MATERIALIZED VIEW v AS VALUES ...`) would do the trick?

**Comment by ruchirK** [comment]
I'm pretty sure that tailing a static view indeed is the repro 

```
materialize=> create materialized view foo (a) as values (1), (2), (3);
CREATE VIEW
materialize=> tail foo;
1	Diff: 1 at 18446744073709551615
2	Diff: 1 at 18446744073709551615
3	Diff: 1 at 18446744073709551615
materialize=> tail foo as of 5;
1	Diff: 1 at 5
2	Diff: 1 at 5
3	Diff: 1 at 5
materialize=>
```

this makes sense because sinks have tests like that and tails do not, `18446744073709551615` is uint64_max and I double checked with a printline.

sidenote: is the thing where it reports the diffs as happening at 5 if we request as_of 5 expected behavior? I am not sure offhand, will look more closely tomorrow

**Comment by frankmcsherry** [comment]
> is the thing where it reports the diffs as happening at 5 if we request as_of 5 expected behavior?

I believe so. The diffs actually happen at time `0`, and in each case the `AS OF` advances it to that time.

**Comment by ruchirK** [comment]
right right. that makes sense. I keep forgetting that this output doesn't necessarily mean that the diff "occurred at" that time, but it reflects the state of the collection at that time. I wonder if there's a way to visually make the distinction between the diffs at the snapshot vs diffs that come in at real time more clear

**Comment by frankmcsherry** [comment]
> would be nice to remember original timestamps for records that have been advanced and print the original time too

Part of the problem is that if we want to do this, we don't get compaction. That information needs to be forgotten at some point.

> I wonder if there's a way to visually make the distinction between ...

I think morally the thing we should be doing that we are not is tracking `as_of` metadata about our tails and sinks. Each one should communicate their `as_of` frontier, so that users don't mistakenly treat the data as accurate at prior times. It would also indicate which times are in the compacted updates.

**Comment by JLDLaughlin** [comment]
Closed with #3500
